src/com/facebook/buck/jvm/scala/ScalacStep.java

```diff
@@ -88,8 +97,24 @@ public String getShortName() {
     } else {
       commandBuilder.add("-classpath", classpath);
     }
+
+    ImmutableList<Path> expandedSources;
+    try {
+      expandedSources = ExternalJavac.getExpandedSourcePaths(
+          filesystem,
+          invokingRule,
+          sourceFilePaths,
+          Optional.of(workingDirectory),
+          ".scala");
+    } catch (IOException e) {
+      throw new HumanReadableException(
+          "Unable to expand sources for %s into %s",
+          invokingRule,
+          workingDirectory);
+    }
+
```

Comment:
I saw that `ExternalJavac` supports passing things off via `@` listfiles. That might be something to support in Scala as well.